tools/proto2jsonschema/postgen.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+const recursive = require("recursive-readdir");
+const camelcaseKeys = require('camelcase-keys');
+
+/**
+ * This tool polishes the JSON schemas with a few modifications:
+ * - Adds "$id" – "." delimited ID – e.g. "google.events.cloud.audit.v1.LogEntryData"
+ * - Adds "name" – The name for the JSON schema – e.g. "LogEntryData"
+ * - Removes snake_case fields that are duplicative of camelCase fields that are seen in JSON payloads
+ *   - This is a feature request for the proto2jsonschema generator.
+ */
+const ROOT = path.resolve(`${__dirname}/../../jsonschema`);
+console.log(`Fixing paths in dir: ${ROOT}`);
+(async () => {
+  const filePaths = await recursive(ROOT);
+  // For every file
+  filePaths.map(filePath => {
+    const dataName = path.basename(filePath,  path.extname(filePath)); // i.e. LogEntryData
+    
+    // Create the modified JSON schema output
+    const json = JSON.parse(fs.readFileSync(filePath).toString());
+    const resultJSON = {
+      // Add the name and $id first
+      name: dataName,
+      $id: getId(filePath),
+      // Add all other fields. Convert keys to camelCase (i.e. remove snake_case keys)
+      ...camelcaseKeys(json, {deep: true})
```

Comment:
nit: Slight performance optimization inserting the name and $id into the output of camelcaseKeys instead of copying the existing schema into this new object.

Reply:
There isn't much performance optimization for not copying the JSON file, especially at the sacrifice of readability.

I'm seeing this time for the whole postgen script:

```
real	0m0.107s
user	0m0.108s
sys	0m0.025s
```

I think optimizing should come at >10 seconds, and these <100 fs reads are probably not the first place to optimize.